File: frontend_app/src/lib/archetypes.ts
import type { Archetype } from "./api";

/**
 * The four archetypes, mirroring services/Orchestration/agents/archetypes.py
 * and frontend/src/lib/content.ts. The Orchestrator is deliberately absent: it
 * is a system role the engine supplies, never assigned to an entity.
 */
export const ARCHETYPES: { id: Archetype; name: string; role: string }[] = [
  { id: "strategist", name: "The Strategist", role: "Plans forward" },
  { id: "skeptic", name: "The Skeptic", role: "Tests the claim" },
  { id: "loyalist", name: "The Loyalist", role: "Defends the position" },
  { id: "wildcard", name: "The Wildcard", role: "Breaks the frame" },
];

/** Two-letter code for the node badge and the picker. */
export const ARCHETYPE_CODE: Record<Archetype, string> = {
  strategist: "St",
  skeptic: "Sk",
  loyalist: "Lo",
  wildcard: "Wi",
};

export const ARCHETYPE_BY_ID = Object.fromEntries(
  ARCHETYPES.map((a) => [a.id, a]),
) as Record<Archetype, { id: Archetype; name: string; role: string }>;

File: services/Orchestration/agents/archetypes.py
"""
The four archetypes an entity can be promoted into.

These are BEHAVIOURAL SETTINGS, not characters. None of them belongs to any
particular entity: the same archetype can be worn by different entities, and
one entity can be re-promoted under a different one. What the human picks here
is what decides how the agent argues once it is talking.

The Orchestrator on the marketing page is deliberately NOT in this map. It is a
system role the engine supplies, one per run, and it is never assigned to an
entity. Keeping it out of the data means it cannot be offered as a fifth choice
by accident.

The ids match frontend/src/lib/content.ts so the console and the backend agree
on the same four names.
"""

ARCHETYPES: dict[str, dict] = {
    "strategist": {
        "name": "The Strategist",
        "role": "Plans forward",
        "persona": (
            "You read the relationships around your entity for leverage and "
            "propose the sequence of moves you think survives contact with the "
            "other parties. You think in timing and dependencies, not just in "
            "positions."
        ),
    },
    "skeptic": {
        "name": "The Skeptic",
        "role": "Tests the claim",
        "persona": (
            "You attack the weakest assumption in whatever was just said, and "
            "ask where the evidence actually came from. You are not contrarian "
            "for sport: you want the claim that survives the challenge."
        ),
    },
    "loyalist": {
        "name": "The Loyalist",
        "role": "Defends the position",
        "persona": (
            "You argue from your entity's stated interest and supply the "
            "details only someone on the inside of it would carry. You defend "
            "your side first and concede only what the record forces you to."
        ),
    },
    "wildcard": {
        "name": "The Wildcard",
        "role": "Breaks the frame",
        "persona": (
            "You ignore the premise everyone else agreed on and surface the "
            "outcome nobody in the room priced in. You are not random; you are "
            "the option the group's shared assumptions made invisible."
        ),
    },
}

# Order matters for anything that lists them; this is the menu order.
ARCHETYPE_IDS = tuple(ARCHETYPES)
